### ISSUE TITLE
Skyrim Special Edition: Creation Kit (1946180)

### ISSUE THREAD

**Issue by kgzod** [ValveSoftware/Proton#7162]
# Compatibility Report
- Name of the game with compatibility issues: Skyrim Special Edition: Creation Kit Quest "Dialogue Views" and "Scenes" menus are broken.
- Steam AppID of the game: 1946180

## System Information
- GPU: GTX 1070
- Video driver version: nvidia-driver-535 Version 535.113.01-0ubuntu0.22.04.1
- Kernel version: 5.15.0-86-generic
- System Info: https://gist.github.com/king-zoidberg/00e3f387baca0c2396c74b419bdaa1a2
- Proton version: 8.0-4 (same problem on all other available full releases as well as hotfix and experimental as of oct 9 2023)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

- Proton Log: https://gist.github.com/king-zoidberg/fddbfeea4b7d5859ceb5bf940c5d7488

## Symptoms

The interactive flowchart used by the creation kit in the Scenes and Dialogue Views is not drawn (only a while box shows up in its place) and cannot be interacted with by right clicking (nothing happens on right click). Both the original Skyrim: Creation Kit (for the original release of the game not special edition) and the Fallout 4: Creation Kit correctly draw the flowchart and can be interacted with on proton 8.0-4.

## Reproduction

Load the base game data Skyrim.esm open a quest with a dialogue view or scene select the dialogue view or scene and the flowchart box on the right of the window remains unchanged.

**Comment by Blayer98** [comment]
Hey, I wanted to ask if there were any issues with saving plugins on Skyrim’s Creation Kit, as that’s what’s happening with Starfield’s CK. #7884

I haven’t played around with the CK for Starfield regarding the flowcharts issue to see if it still happens there though. That’s if that feature is still in Starfield’s CK, I’ll try installing Skyrim and it’s CK and see if anything’s changed there.

**Comment by Blayer98** [comment]
[steam-1946180.log](https://github.com/user-attachments/files/16196724/steam-1946180.log)

So from what I’ve noticed, there appears to be a random crash happening, not sure why or exactly how yet and unfortunately didn’t get that captured in this log, but I did test out the Scene and Dialogue view, and found that the issue regarding the flowcharts still persists on the latest Proton Experimental. (9.0-2) It seems like it’s known to be buggy however from this reddit post I found: https://www.reddit.com/r/skyrimmods/s/Hqvsv2G5GZ

**Comment by poperigby** [comment]
I'm noticing an issue where the keyboard controls in the render view seem to get "stuck" sometimes, where it will keep moving with my mouse, or move too much when I'm panning/orbiting. Using Proton Experimental.

**Comment by GammaMetroid** [comment]
I'm encountering a crash nearly every time I try to assign a texture set to a model. There were a couple times that it did work correctly, but I haven't been able to reproduce that. I never got a crash this way on Windows. In my proton log, I can see an access violation on line 24097 when I assign the texture set:

[steam-1946180.log](https://github.com/user-attachments/files/24065128/steam-1946180.log)

It looks like at least one other user has experienced this same thing: https://forums.nexusmods.com/topic/13526477-creation-kit-crashes-when-assigning-texture-to-models/

**Comment by kisak-valve** [comment]
Skyrim SE Creation Kit

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9571.
@Arthmoor posted on 2026-03-13T22:52:22:

# Compatibility Report
Skyrim Special Edition: Creation Kit
Steam AppID: 1946180

## System Information
- GPU: Radeon 9070 XT
- Video driver version: Mesa 25.3.6
- Kernel version: 6.18.16-200.fc43.x86_64
- Proton version: Proton Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms
The scene editor for quests displays a blank window instead of the expected information.

There is a DLL file called flowchartx64.dll which is involved in the operation of the scene editing window. This is apparently some sort of ActiveX/COM component that ships with the Creation Kit on Skyrim, Fallout 4, and Starfield. I don't know if the files are functionally different with each game, but I know for sure the one for Skyrim SE does not work.

When the Creation Kit is first installed it asks if I want to register this DLL file. I say yes to the prompt. It either is not being registered properly in the Proton instance or the DLL itself is somehow not being processed properly.


## Reproduction

Skyrim Special Edition first needs to be installed by Steam.
Then you can install the Skyrim Special Edition: Creation Kit afterward.
Once it is installed, start the Creation Kit.
You should get a prompt about the Scripts.zip file, which can be safely ignored.
You should also have another prompt to register the flowchartx64.dll file. Say yes to this prompt.

When the CK finishes loading, hit the "Open File" button. Select Skyrim.esm from the file menu. Double click on it, then hit OK to start it.
It will take some time to load its information.

When done, open the "Caravans" entry from the list of quests.
Click the Scenes tab.
Select the one listed in the menu.
At this point you should see that the scene window is empty.

What you should see instead is something like this:
https://variedvoices.weebly.com/uploads/1/1/7/0/1170875/untitled5_6_orig.png

**Comment by Arthmoor** [comment]
I'd like to add that in the Fallout 4 Creation Kit, the exact same flowchartx64.dll file works just fine with that kit's scene editor.

The same is true for Starfield's Creation Kit. This exact same .dll works just fine there.

**Comment by kgzod** [comment]
So I figured out the mechanism of this bug around a year ago and forgot to ever come back here to explain. 

TLDR it's probably not able to be fixed in wine/proton itself.

When setting up the menu the creation kit calls a function in flowchartx64.dll at 0x180070d40 as far as I have been able to tell looking at the disassembly this is a function that makes sure the menu is drawn at a resolution smaller than the screen so it fits inside. It is broken and never initializes the rect struct that holds the resolution to compare against on all versions of wine the stack has values in that makes the resolution be read as 1, if you get really close to your screen and line up your mouse at the top of where the menu should be you are able to interact with it. It is being drawn and it works but with a vertical resolution of 1. On Windows the values left in the uninitialized memory are always read as negative since the menu cant have a resolution the function always returns NULL on Windows and a later function in flowchartx64.dll checks to make sure the menu was created, it never is at this point so it just creates the menu with the default resolution regardless of weather or not it will fit on the screen, that works and that is the menu seen on Windows. So I spent a while trying to figure out why the values in the stack would be different on Windows and wine and found that the stack on Windows changed after the kernel call NtUserShowWindow. I found a write up about a Windows kernel security vulnerability that explained what was going on here (source https://media.blackhat.com/bh-us-11/Mandt/BH_US_11_Mandt_win32k_WP.pdf) at some point around Windows Vista Microsoft moved large amounts of the window management kernel calls out of the kernel but to avoid breaking existing software kept the kernel calls in ntdll.dll. So on windows NtUserShowWindow still traps into the kernel but when returning from kernel mode to user mode the thread has a new stack frame and is executing the KiUserCallbackDispatcher function that serves as an inter-process communication function that tells the window manager what to do. The arguments to KiUserCallbackDispatcher is what is initializing the stack memory with data that happens to make the menu show up correctly. Since wine does not edit the "user" stack and switches to a linux stack when running windows kernel calls this will never happen on wine.  I did ask around on the wine gitlab and the opinion was that this is not fixable in wine. Also as a side this same menu bug could appear on windows if Microsoft changes the any function used by the creation kit moving stack memory around or changes the undocumented functionality of KiUserCallbackDispatcher when calling GUI functions, it's a small miracle this hasn't broken on Windows. So since this is a problem with flowchartx64.dll I did (badly) make a patcher program here https://github.com/kgzod/SkyrimSE-CreationKit-Wine-Patch/tree/main that just removes the body of the broken function so it always fails just like on Windows.
